StRoot/StGmtAligner/EventT.cxx

```diff
@@ -0,0 +1,551 @@
+       << "\tNpoint " << fNpoint << "\tNsp " << fNsp << endl;
+  for (UInt_t i = 0; i < fNsp; i++) cout << "\t" << fIdHitT[i];
+  cout << endl;
+}
+//________________________________________________________________________________
+void HitT::SetId(Int_t B, Int_t L, Int_t l, Int_t W, Int_t H) {
+  barrel = B; layer = L; ladder = l; wafer = W; hybrid = H;
+}
+//________________________________________________________________________________
+void HitT::Print(Option_t *opt) const {
+  cout << "HitT: Id " << Id << "\tpT = " << pT << "\tmomentum " << pMom << endl;
+  TRVector glob(3,&xG); cout << "Global :" << glob << endl;
+  cout << "Local      u/v/w " << u << "/ " << v << "/ " << w << endl;
+  cout << "Prediction uP/vP " << uP << "/ " << vP << "\ttuP/tvP " << tuP << "/ " << tvP << endl;
+}
+//________________________________________________________________________________
+void EventT::RestoreListOfRotations() {
+  if (fRotList) return;
+  if (! gDirectory) return;
+  fRotList = new THashList(100,0);
+  fRotList->SetOwner();
+  TIter nextkey(gDirectory->GetListOfKeys() );
+  TKey *key;
+  while ((key = (TKey*) nextkey())) {
+    TObject *obj = key->ReadObj();
+    if ( obj->IsA()->InheritsFrom( "TGeoHMatrix" ) ) {
+      fRotList->Add(obj);
+    }
+  }
+}
+//________________________________________________________________________________
+void TBase::Loop(Int_t Nevents) {  
+#if 1
+  struct PlotPar_t {
+    Char_t *Name;
+    Char_t *Title;
+    Int_t    nx;
+    Int_t    ny;
+    Double_t xmin; 
+    Double_t xmax; 
+    Double_t ymin;
+    Double_t ymax;  
+  };
+  const  PlotPar_t plotUP = // plots for uP
+    { "uP","track u", 320, 3, -5.,5., 0.,3. };
+  
+  const  PlotPar_t plotDu = // plots for u-uP
+    { "Du","Du before cut", 250, 3, -2.,2., 0.,3. };
+  
+  const  PlotPar_t plotDuv = // plots for du & dv
+    { "Du","Du cuts", 200, 3, -1.,1., 0.,3. };
+  
+  TFile *fOut = new TFile(fOutFileName,"recreate");
+  TString Name;
+  TString Title;
+  TString uName;
+  TString uTitle;
+  enum {NM = 8}; // no. of modules
+  //              B 
+  TH1F *LocPlots[NM];
+  TH1F *  uPlots[NM];
+  TH1F *    hpT = new TH1F(   "Pt",   "pt", 200, -2., 2.);
+  TH1F *    hpM = new TH1F( "Ptot", "ptot", 200,  0., 5.);
+  TH1F * uPAll  = new TH1F("UPall","uPall", plotUP.nx, plotUP.xmin, plotUP.xmax);
+  TH1F * uAll   = new TH1F("Uall", "ua",    plotUP.nx, plotUP.xmin, plotUP.xmax);
+  TH1F * duB[NM][2];
+  TH1F * dvB[NM];
+  TH1F * uCuts[NM];
+  TH1F * xCuts[NM];
+  TH1F * uCut   = new TH1F("Ucut","uc", plotDu.nx, plotDu.xmin, plotDu.xmax);
+  TH1F * vCut   = new TH1F("Vcut","vc", 200, -3., 3.);
+  TH2F * dMin   = new TH2F("DMin","vumin",100,-0.75,0.75,100,-0.75,0.75); 
+  TH1F * vMin   = new TH1F("VMin","vmin", plotDuv.nx, plotDuv.xmin, plotDuv.xmax);
+  TH1F * uMin   = new TH1F("UMin","umin", plotDuv.nx, plotDuv.xmin, plotDuv.xmax);
+  TH1F * uMinC  = new TH1F("UMinC","umC", plotDuv.nx, plotDuv.xmin, plotDuv.xmax);
```

Comment:
How is this histogram used? The compiler complains about the variable not being used